PyStemmusScope/soil_io.py

```diff
@@ -0,0 +1,229 @@
+        ptf_files, lat, lon, lat_key='latitude', lon_key='longitude'
+    )
+
+    depths = [0, 5, 30, 60, 100, 200]
+
+    alpha = np.zeros(len(depths))
+    Ks = np.zeros(len(depths))
+    theta_s = np.zeros(len(depths))
+    theta_r = np.zeros(len(depths))
+    coef_n = np.zeros(len(depths))
+
+    for i, depth in enumerate(depths):
+        alpha[i] = ds[f"alpha_{depth}cm"]
+        Ks[i] = ds[f"Ks_{depth}cm"]
+        theta_s[i] = ds[f"thetas_{depth}cm"]
+        theta_r[i] = ds[f"thetar_{depth}cm"]
+        coef_n[i] = ds[f"n_{depth}cm"]
+
+    hydraulic_matfiledata = {
+        'SaturatedMC': theta_s,
+        'ResidualMC': theta_r,
+        'Coefficient_n': coef_n,
+        'Coefficient_Alpha': alpha,
+        'porosity': theta_s,
+        'Ks0': Ks[0],
+        'SaturatedK': vc.per_day_to_per_second(Ks),
+        'fieldMC': vc.field_moisture_content(theta_r, theta_s, alpha, coef_n),
+        'theta_s0': theta_s[0]
+    }
+
+    return hydraulic_matfiledata
+
+
+def _read_surface_data(soil_data_path, lat, lon):
+    """Internal function that reads the fmax variable from the surface dataset and
+    returns it in a dictionary.
+
+    Args:
+        soil_data_path (Path): Path to the directory which contains the surface data.
+        lat (float): Latitude of the site of interest (in degrees North)
+        lon (float): Longitude of the site of interest (in degrees East)
+
+    Returns:
+        dict: Dictionary containing the `fmax` value (maximum fractional saturated area)
+    """
+    ds = xr.open_dataset(soil_data_path / 'surfdata.nc')
+    ds = ds.sel(
+        lsmlat=vc.lat_to_lsmlat(lat), lsmlon=vc.lon_to_lsmlon(lon))
+
+    fmax = ds['FMAX'].values
+
+    surface_matfiledata = {'fmax': fmax}
+
+    return surface_matfiledata
+
+
+def _collect_soil_data(soil_data_path, lat, lon):
+    """Internal function that calls the individual data collectors and merges them into
+    a single dictionary ready to be written.
+
+    Args:
+        soil_data_path (Path): Path to the directory which contains the soil data.
+        lat (float): Latitude of the site of interest (in degrees North)
+        lon (float): Longitude of the site of interest (in degrees East)
+
+    Returns:
+        dict: Dictionary containing all the processed soil property data.
+    """
+
+    matfiledata = {}
+
+    lambda_directory = soil_data_path / 'lambda'
+
+    matfiledata.update(_read_lambda_coef(lambda_directory, lat, lon))
+    matfiledata.update(_read_hydraulic_parameters(soil_data_path, lat, lon))
+    matfiledata.update(_read_soil_composition(soil_data_path, lat, lon))
+    matfiledata.update(_read_surface_data(soil_data_path, lat, lon))
+
+    return matfiledata
+
+
+def _retrieve_latlon(plumber2_file):
+    """Retrieves the latitude and longitude coordinates from the PLUMBER2 dataset file,
+    to allow for the
+
+    Args:
+        plumber2_file (Path): Full path to the netCDF file containing the site latitude
+            and longitude
```

Comment:
Let's remove "**plumber2**" from the function and make it generic as much as possible. This allows using datasets from other sources in the future, e.g. era5 data.